src/Docker/Client/Api.hs

```diff
@@ -93,6 +94,11 @@ startContainer sopts cid = requestUnit POST $ StartContainerEndpoint sopts cid
 stopContainer :: forall m. Monad m => Timeout -> ContainerID -> DockerT m (Either DockerError ())
 stopContainer t cid = requestUnit POST $ StopContainerEndpoint t cid
 
+-- | Blocks until a container with the given 'ContainerID' stops,
+-- then returns the exit code
+waitContainer :: forall m. Monad m => ContainerID -> DockerT m (Either DockerError ())
```

Comment:
The doc string seems to suggest that this shall return an exit code but the type signature says otherwise. Do we want to return and exit code here? cc @jprider63